daemon/tests/harness/mocks/mod.rs

```diff
@@ -32,6 +32,38 @@ impl Mocks {
     pub async fn oracle(&mut self) -> MutexGuard<'_, oracle::MockOracle> {
         self.oracle.lock().await
     }
+
+    // Helper function setting up a "happy path" wallet mock
+    pub async fn mock_wallet_sign_and_broadcast(&mut self) {
+        let mut seq = mockall::Sequence::new();
+        self.wallet()
+            .await
+            .expect_sign()
+            .times(1)
+            .returning(|_| Ok(dummy_partially_signed_transaction()))
+            .in_sequence(&mut seq);
+        self.wallet()
+            .await
+            .expect_broadcast()
+            .times(1)
+            .returning(|_| Ok(dummy_tx_id()))
+            .in_sequence(&mut seq);
+    }
+
+    pub async fn mock_oracle_annoucement(&mut self) {
+        self.oracle()
+            .await
+            .expect_get_announcement()
+            .return_const(Some(oracle::dummy_announcement()));
+    }
```

Comment:
One reason I am liking this is because we can easily make `dummy_announcement` configurable if we need at some point!